enhancements/monitoring/replace-prometheus-adapter-with-metrics-server.md

```diff
@@ -0,0 +1,264 @@
+
+1. **Scability:** Starting from `v0.5.0` Metrics Server comes with default resource requests that guarantee good performance for most cluster configurations and it continues to do [scale testing](https://github.com/kubernetes-sigs/metrics-server/blob/077a462cd660beef705afb0a4a14a707ef39c4f2/FAQ.md#how-large-can-clusters-be) which is not the case for Prometheus Adapter.
+
+### User Stories
+
+#### Story 1
+
+* In order to provide [optional built-in monitoring](https://issues.redhat.com/browse/MON-3152) feature we need to provide a way to support HPA and VPA without a Prometheus-based solution.
+
+#### Story 2
+
+* As a cluster admin, I seek a lightweight and efficient solution with improved accuracy for monitoring resource metrics within my cluster.
+
+#### Story 3
+
+* As a cluster admin, I want a simplified monitoring architecture in OpenShift, so that I can reduce the complexity and maintenance overhead of the monitoring system.
+
+### Goals
+
+* Replace Prometheus Adapter with Metrics Server while maintaining feature parity with the existing stack 
+* Ensure no performance regression occurs when replacing Prometheus Adapter with Metrics Server
+* Improved scale up time for autoscaling
+
+### Non-Goals
+
+* Allow optional cluster monitoring stack
+* Moving `openshift-monitoring/cluster-monitoring-config` config-map to a Custom Resource
+
+## Proposal
+
+The proposal is to replace the Prometheus Adapter with the Metrics Server for monitoring resource metrics in OpenShift.
+
+### Workflow Description
+
+The workflow for using the Metrics Server in OpenShift for monitoring resource metrics would involve the following steps:
+
+1. User enables `TechPreviewNoUpgrade` featureset
+
+    ```yaml
+    apiVersion: config.openshift.io/v1
+    kind: FeatureGate
+    metadata:
+      name: cluster
+    spec:
+      featureSet: TechPreviewNoUpgrade
+    ```
+
+    With this configuration, the cluster monitoring operator should:
+    1. Deploy the `metrics-server` resources in the `openshift-monitoring` namespace
+    1. Verify that the `metrics-server` service is functional by querying the `/readyz` endpoint
+    1. Wait for `5 x metrics-server's scrape interval` to have enough data populated in `metrics-server` in-memory
+    1. Update `APIService` resource to point to `metrics-server` API instead of `prometheus-adapter` API
+    1. Delete `prometheus-adapter` resources
+
+### HyperShift Assumptions
+
+1. The existing resource metrics API implementation, prometheus-adapter, is deployed on the guest cluster's data plane, so will be metrics-server, because it requires kubelet scraping. Consequently, we don't anticipate any effects on Hypershift due to this enhancement.
```

Comment:
(nit)
```suggestion
The existing resource metrics API implementation, prometheus-adapter, is deployed on the guest cluster's data plane by CMO which is itself deployed by the Cluster Version operator. The same will be true for the Metrics server because it requires kubelet scraping. Consequently, we don't anticipate any effects on Hypershift as part of this enhancement.
```